src/stats/stats.jl

```diff
@@ -0,0 +1,158 @@
+export ExecutionStats, statshead, statsline
+
+const STATUS = Dict(:unknown => "unknown",
+                    :first_order => "first-order stationary",
+                    :max_eval => "maximum number of function evaluations",
+                    :max_time => "maximum elapsed time",
+                    :max_iter => "maximum iteration",
+                    :neg_pred => "negative predicted reduction",
+                    :unbounded => "objective function may be unbounded from below",
+                    :exception => "unhandled exception"
+                   )
+
+type ExecutionStats
+  status :: Symbol
+  solved :: Bool
+  tired :: Bool
```

Comment:
Same question about `:tired`. Isn't it the same as `status in [:max_eval, :max_time, :max_iter]`?

Reply:
Since we can have tired and solved, I felt the need to include `solved`, `tired` and `stalled` in ExecutionStats. But I do actually prefer to rely solely on `status`.
Maybe I could remove them altogether?

Reply:
That would seem simpler to me.